packages/profile/src/profile.ts

```diff
@@ -338,10 +338,17 @@ export function resolveZoneFileToProfile(zoneFile: any, publicKeyOrAddress: stri
     } catch (e) {
       reject(e);
     }
-
+    let isProfileUrl: boolean = false;
+    try {
+      // Check if zoneFile is only a profile url
```

Comment:
This always happens for sub domains, correct? Should be in the comment because it is not clear from the naming.

Reply:
> Looks good, but the code is not easy to understand why the zone file can be the profile url.
Update: 
**Explanation**:  Below is the examples where you can see that zone file is only a profile url.
**Url**: https://stacks-node-api.mainnet.stacks.co/v1/names/jfontirroig.xck.app
**Response**:
```
{
  "address": "SP31C1RGYVJVWNYWJNRVSZQA2XN9HDEA08FB0G03S",
  "blockchain": "stacks",
  "last_txid": "0x0b80143fae8a52f47c970c5cf935ae37cd13236887e2a470d8a6891091219986",
  "resolver": "",
  "status": "registered_subdomain",
  "zonefile": "https://gaia.blockstack.org/hub/1BhQxPY3N9ABXzkbr4U8J6gkvAdr9Acr8L/profile.json",
  "zonefile_hash": "fdccd9ca23ec81f8d5ef0057eb2f72fb"
}
```
That's why i added this check to ensure that if its a profile url then do not run JSON.parse as it fails because its not a zoneFile object. 
So this check will skip the JSON.parse and retrieve the profile using the url mentioned in the zonefile attribute.  I also updated the comment to explain this point in code.